### PR TITLE
[test] Improve Papaya samples

### LINKED ISSUE #4064: Allow project template images to fill project template cards
As visible in the screenshot below, the cards blank project or upload project can have a nice background (here a solid color) while the other project templates have an image centered with a ton of padding which is quite ugly.

Capture d’écran 2024-10-03 à 17 08 05

### PR DESCRIPTION
**Pull request by sbegaudeau**
- **[4064] Ensure that project template images fill the whole content part of the card**
- **[test] Improve the Papaya class diagram with new tools**
- **[test] Separate Papaya examples in various project samples**

# Pull request template

## General purpose
What is the main goal of this pull request?
- [ ] Bug fixes
- [ ] New features
- [ ] Documentation
- [ ] Cleanup
- [ ] Tests
- [ ] Build / releng

## Project management
- [ ] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [ ] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [ ] Have the relevant issues been added to the pull request?
- [ ] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [ ] Have the relevant issues been added to the same project and milestone as the pull request?
- [ ] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

File: packages/sirius-web/backend/sirius-web-papaya/src/main/java/org/eclipse/sirius/web/papaya/representations/classdiagram/tools/diagram/ClassDiagramDropToolProvider.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2024 Obeo.
+ * Copyright (c) 2024, 2024 Obeo.
```
Comment (frouene) [inline packages/sirius-web/backend/sirius-web-papaya/src/main/java/org/eclipse/sirius/web/papaya/representations/classdiagram/tools/diagram/ClassDiagramDropToolProvider.java]: Wrong dates

File: packages/sirius-web/backend/sirius-web-papaya/src/main/java/org/eclipse/sirius/web/papaya/representations/classdiagram/tools/diagram/ImportExistingTypesToolProvider.java [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2024 Obeo.
+ * Copyright (c) 2024, 2024 Obeo.
  * This program and the accompanying materials
```
Comment (frouene) [inline packages/sirius-web/backend/sirius-web-papaya/src/main/java/org/eclipse/sirius/web/papaya/representations/classdiagram/tools/diagram/ImportExistingTypesToolProvider.java]: wrong dates

File: packages/sirius-web/backend/sirius-web-papaya/src/main/java/org/eclipse/sirius/web/papaya/representations/classdiagram/tools/diagram/NewClassToolProvider.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
+/*******************************************************************************
+ * Copyright (c) 2024, 2024 Obeo.
+ * This program and the accompanying materials
```
Comment (frouene) [inline packages/sirius-web/backend/sirius-web-papaya/src/main/java/org/eclipse/sirius/web/papaya/representations/classdiagram/tools/diagram/NewClassToolProvider.java:3]: wrong dates

File: packages/sirius-web/backend/sirius-web-papaya/src/main/java/org/eclipse/sirius/web/papaya/representations/classdiagram/tools/diagram/NewEnumToolProvider.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
+/*******************************************************************************
+ * Copyright (c) 2024, 2024 Obeo.
+ * This program and the accompanying materials
```
Comment (frouene) [inline packages/sirius-web/backend/sirius-web-papaya/src/main/java/org/eclipse/sirius/web/papaya/representations/classdiagram/tools/diagram/NewEnumToolProvider.java:3]: wrong dates

File: packages/sirius-web/backend/sirius-web-papaya/src/main/java/org/eclipse/sirius/web/papaya/representations/classdiagram/tools/diagram/NewInterfaceToolProvider.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
+/*******************************************************************************
+ * Copyright (c) 2024, 2024 Obeo.
+ * This program and the accompanying materials
```
Comment (frouene) [inline packages/sirius-web/backend/sirius-web-papaya/src/main/java/org/eclipse/sirius/web/papaya/representations/classdiagram/tools/diagram/NewInterfaceToolProvider.java:3]: wrong dates

File: packages/sirius-web/backend/sirius-web-papaya/src/main/java/org/eclipse/sirius/web/papaya/representations/classdiagram/tools/diagram/NewRecordToolProvider.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
+/*******************************************************************************
+ * Copyright (c) 2024, 2024 Obeo.
+ * This program and the accompanying materials
```
Comment (frouene) [inline packages/sirius-web/backend/sirius-web-papaya/src/main/java/org/eclipse/sirius/web/papaya/representations/classdiagram/tools/diagram/NewRecordToolProvider.java:3]: wrong dates

File: packages/sirius-web/backend/sirius-web-papaya/src/main/java/org/eclipse/sirius/web/papaya/representations/classdiagram/tools/interfacenode/ImportAllInterfaceSubtypesNodeToolProvider.java [hunk 7]
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2024 Obeo.
+ * Copyright (c) 2024, 2024 Obeo.
  * This program and the accompanying materials
```
Comment (frouene) [inline packages/sirius-web/backend/sirius-web-papaya/src/main/java/org/eclipse/sirius/web/papaya/representations/classdiagram/tools/interfacenode/ImportAllInterfaceSubtypesNodeToolProvider.java]: wrong dates

File: packages/sirius-web/backend/sirius-web-papaya/src/main/java/org/eclipse/sirius/web/papaya/representations/classdiagram/tools/interfacenode/ImportInterfaceImplementationsNodeToolProvider.java [hunk 8]
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2024 Obeo.
+ * Copyright (c) 2024, 2024 Obeo.
  * This program and the accompanying materials
```
Comment (frouene) [inline packages/sirius-web/backend/sirius-web-papaya/src/main/java/org/eclipse/sirius/web/papaya/representations/classdiagram/tools/interfacenode/ImportInterfaceImplementationsNodeToolProvider.java]: wrong dates

**Review by frouene — APPROVED** [review]

File: packages/sirius-web/backend/sirius-web-papaya/src/main/java/org/eclipse/sirius/web/papaya/representations/classdiagram/tools/interfacenode/ImportInterfaceImplementationsNodeToolProvider.java (hunk 8 above)
Comment (sbegaudeau) [reply packages/sirius-web/backend/sirius-web-papaya/src/main/java/org/eclipse/sirius/web/papaya/representations/classdiagram/tools/interfacenode/ImportInterfaceImplementationsNodeToolProvider.java]: Done

File: packages/sirius-web/backend/sirius-web-papaya/src/main/java/org/eclipse/sirius/web/papaya/representations/classdiagram/tools/interfacenode/ImportAllInterfaceSubtypesNodeToolProvider.java (hunk 7 above)
Comment (sbegaudeau) [reply packages/sirius-web/backend/sirius-web-papaya/src/main/java/org/eclipse/sirius/web/papaya/representations/classdiagram/tools/interfacenode/ImportAllInterfaceSubtypesNodeToolProvider.java]: Done

File: packages/sirius-web/backend/sirius-web-papaya/src/main/java/org/eclipse/sirius/web/papaya/representations/classdiagram/tools/diagram/NewRecordToolProvider.java (hunk 6 above)
Comment (sbegaudeau) [reply packages/sirius-web/backend/sirius-web-papaya/src/main/java/org/eclipse/sirius/web/papaya/representations/classdiagram/tools/diagram/NewRecordToolProvider.java:3]: Done

File: packages/sirius-web/backend/sirius-web-papaya/src/main/java/org/eclipse/sirius/web/papaya/representations/classdiagram/tools/diagram/NewInterfaceToolProvider.java (hunk 5 above)
Comment (sbegaudeau) [reply packages/sirius-web/backend/sirius-web-papaya/src/main/java/org/eclipse/sirius/web/papaya/representations/classdiagram/tools/diagram/NewInterfaceToolProvider.java:3]: Done

File: packages/sirius-web/backend/sirius-web-papaya/src/main/java/org/eclipse/sirius/web/papaya/representations/classdiagram/tools/diagram/NewEnumToolProvider.java (hunk 4 above)
Comment (sbegaudeau) [reply packages/sirius-web/backend/sirius-web-papaya/src/main/java/org/eclipse/sirius/web/papaya/representations/classdiagram/tools/diagram/NewEnumToolProvider.java:3]: Done

File: packages/sirius-web/backend/sirius-web-papaya/src/main/java/org/eclipse/sirius/web/papaya/representations/classdiagram/tools/diagram/NewClassToolProvider.java (hunk 3 above)
Comment (sbegaudeau) [reply packages/sirius-web/backend/sirius-web-papaya/src/main/java/org/eclipse/sirius/web/papaya/representations/classdiagram/tools/diagram/NewClassToolProvider.java:3]: Done

File: packages/sirius-web/backend/sirius-web-papaya/src/main/java/org/eclipse/sirius/web/papaya/representations/classdiagram/tools/diagram/ImportExistingTypesToolProvider.java (hunk 2 above)
Comment (sbegaudeau) [reply packages/sirius-web/backend/sirius-web-papaya/src/main/java/org/eclipse/sirius/web/papaya/representations/classdiagram/tools/diagram/ImportExistingTypesToolProvider.java]: Done

File: packages/sirius-web/backend/sirius-web-papaya/src/main/java/org/eclipse/sirius/web/papaya/representations/classdiagram/tools/diagram/ClassDiagramDropToolProvider.java (hunk 1 above)
Comment (sbegaudeau) [reply packages/sirius-web/backend/sirius-web-papaya/src/main/java/org/eclipse/sirius/web/papaya/representations/classdiagram/tools/diagram/ClassDiagramDropToolProvider.java]: Done

**Pull request merged**